### PR TITLE
VQVAE Model for HipMRI Study

### PR DESCRIPTION
**Pull request by SylviaWwww**
This Pull Request introduces a VQVAE-based generative model for the HipMRI Study on Prostate Cancer using the processed 2D slices.

### CODE REVIEW

**Comment by Uto0op** [comment]
Recognition problem: VQVAE on HipMRI ---- Hard Difficulty

1. Algorithm solves the problem:
Not very clear reconstructed images, with an SSIM over 0.5 -1; No training/validation loss/SSIM plot -0.5 (3.5 Marks)

2. Algorithm demonstrated and functions: 
Good attempt with self-attention (3 Marks)

4. Design:
Modularized design (1 Mark)

5. Commenting:
Sufficient docstrings and comments (1 Mark)

6. Algorithm above Normal Difficulty: (5 Marks)

7. Algorithm is Hard difficulty: (5 Marks)

**Suggestions**:

- In README.md you could briefly justify the hyperparameters chosen for the training process, or any tuning on them (e.g. why ended up choosing LR of 1e-4, what were the other attempts and their implications);
- Images not shown properly on README.md due to incorrect path configuration;
- It's a good attempt to have self-attention in the VQVAE, I agree it could improve the representation of global features, so it might be a good idea to visualize the latent/quantized space :);
- You might also want to make use of the validation set;
- Training/Validation loss/SSIM plots are usually expected to have an idea of how the model performs/converges, I think they can give some clues on how to improve the current performance.

**Comment by YashMitttal** [comment]
Additional suggestions for improvement:

- Although residual blocks and self-attention are excellent choices, the relatively low SSIM scores suggest that the model might need further enhancements.
- Provide a bit more detail on how perceptual loss and other loss functions are implemented. (You can provide a loss plot as well). Also, your readme images need to be presented properly.
- Could you explain more about the choices made for hyperparameters like batch size, learning rate, and the number of epochs?
- You can use a validation set as well. I think it should be helpful in your case.

Also, please make your pull req more descriptive.

**Comment by hanemma7moud** [comment]
@SylviaWwww can you please provide your name and student number

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. </td><td>-2</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. </td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard. VQVAE (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Good Progressive commits. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Acceptable. Images are not showing - possible markup errors.</td><td>-2</td></tr>
	<tr><td></td><td>Markdown used and PDF NOT submitted. PDF not submitted.</td><td>-2</td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). On time submission.</td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is good. </td><td></td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-9</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
No proper splits -2, no plots or training evidence -2, low evidence of solution (plots/loss/quality etc.) -1

**Comment by shakes76** [comment]
Please provide name or ID to assign marks

**Comment by SylviaWwww** [comment]
hi, shakes
Name: Siyao Wu
ID: 48284129

Thanks for your help!

Shakes ***@***.***>于2024年11月19日 周二13:12写道：
